File: tournaments/leftLeavesSum/leftLeavesSum.js
//
// Binary trees are already defined with this interface:
// function Tree(x) {
//   this.value = x;
//   this.left = null;
//   this.right = null;
// }
function leftLeavesSum(t, isLeft = false) {
    if (!t) {
        return 0;
    }

    if (!t.left && !t.right) {
        return isLeft ? t.value : 0;
    }

    return leftLeavesSum(t.left, true) + leftLeavesSum(t.right);
}

File: tournaments/leftLeavesSum/leftLeavesSum.cs
//
// Binary trees are already defined with this interface:
// class Tree<T> {
//   public T value { get; set; }
//   public Tree<T> left { get; set; }
//   public Tree<T> right { get; set; }
// }
int leftLeavesSum(Tree<int> t) {
    var result = 0;

    dfs(t, ref result);

    return result;
}

private void dfs(Tree<int> node, ref int sum, bool isLeft = false) {
    if (null == node) {
        return;
    }

    if (isLeft && null == node.left && null == node.right) {
        sum += node.value;
    }

    dfs(node.left, ref sum, true);
    dfs(node.right, ref sum);
}
